src/modules/utils.ts

```diff
@@ -90,3 +90,21 @@ export function findUp({ what, from }) {
 
   return null;
 }
+
+// set deep property in object
+// args = array of prop-names, last arg = value
+export function setProp(obj, ...args) {
```

Comment:
I see what you did here, but I think a signature of setProp(obj, path, value) would be easier to grasp. What do you think?

Reply:
> I see what you did here

i hope others will see it too ; )

alternative: store the key-path in a separate array

```js
SOURCE_MAP_PROP_MAP = {
  typescript: [ ['compilerOptions', 'sourceMap'], true ],
  scss: [ ['sourceMap'], true ],
};

export function setProp(obj, keyList, val) {
  let i = 0;

  for (; i < keyList.length - 1; i++) {
    const key = keyList[i];

    if (typeof obj[key] !== 'object') {
      obj[key] = {};
    }

    obj = obj[key];
  }

  obj[keyList[i]] = val;
}

const name = 'typescript';
const opts = {};

setProp(opts, ...SOURCE_MAP_PROP_MAP[name]);

if (explicitAndVerboseCodingLevel == 100) {
  const [propPath, value] = SOURCE_MAP_PROP_MAP[name];
  setProp(opts, propPath, value);
}
```
benefit: the function signature is self-explanatory, and needs no comment
i avoided the `Array.isArray(keyList)` vs `typeof keyList == 'string'` switch, to reduce overhead

Reply:
I like that better. Separating the path from the value eases the understandability